photon-ml/src/main/scala/com/linkedin/photon/ml/data/AvroDataReader.scala

```diff
@@ -164,6 +164,7 @@ class AvroDataReader(
     // Infer sql schema from avro schema
     val srcFeatureCols = featureColumnMap.values.flatten.toSet
     val schemaFields = inferSchemaFields(records)
+      .getOrElse(throw new IllegalArgumentException("No records found at given input paths."))
```

Comment:
How about adding the actual path in the error message? In my experience, that speeds up debugging: you have everything right there in one message, no need to look somewhere else to see what the input path is.